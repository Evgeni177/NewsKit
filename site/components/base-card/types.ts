import {CardProps} from 'newskit';

export interface BaseCardProps extends CardProps {
  title?: string;
  children: React.ReactNode;
}
