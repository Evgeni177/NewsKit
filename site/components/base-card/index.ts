export * from './base-card';
export * from './types';
