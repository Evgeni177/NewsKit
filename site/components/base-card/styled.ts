import {styled, getMotionFromTheme, Card} from 'newskit';

export const StyledCard = styled(Card)`
  transition-property: box-shadow;
  transition-duration: ${getMotionFromTheme('motionDuration030')};
  transition-timing-function: ${getMotionFromTheme('motionTimingEaseOut')};
`;
