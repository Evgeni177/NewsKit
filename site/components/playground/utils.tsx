import React from 'react';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const isComponent = (value: any): value is React.ComponentType =>
  Boolean(value && (value.$$typeof || value.displayName));

export const withProps = (
  Component: React.ComponentType,
  props: Record<string, unknown>,
  displayName?: string,
) => {
  const Comp: React.FC = p => <Component {...props} {...p} />;
  Comp.displayName = displayName || Component.displayName;
  return Comp;
};
