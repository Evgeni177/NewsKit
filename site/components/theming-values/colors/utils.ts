import {Theme, ThemeProp} from 'newskit';
import {ThemeColor} from './types';
import {getContrast, getBrightness, getContrastRating} from './color-utils';

export const getBorderRadius = ({
  first,
  last,
  theme,
}: {
  first?: boolean;
  last?: boolean;
  theme?: Theme;
}) => {
  const fn = ({
    theme: {
      borderRadius: {borderRadiusRounded020: radius},
    },
  }: ThemeProp) =>
    [
      first ? radius : 0,
      first ? radius : 0,
      last ? radius : 0,
      last ? radius : 0,
    ].join(' ');

  if (theme) {
    return fn({theme});
  }

  return fn;
};

export const getTextColor = ({
  theme,
  value,
}: // brightness,
{theme: Theme} & Pick<ThemeColor, 'brightness' | 'value'>) =>
  // brightness < 0.4 ? theme.colors.inkInverse : theme.colors.inkContrast;
  getContrast(value, theme.colors.inkInverse) <
  getContrast(value, theme.colors.inkContrast)
    ? theme.colors.inkInverse
    : theme.colors.inkContrast;

export const getColorGroup = (name: string) => (/[a-z]+/.exec(name) || [])[0];

export const mapColorObjects = (
  theme: Theme,
  paletteColorMap?: Record<string, ThemeColor[]>,
) => ([name, value]: [string, string]) => {
  const brightness = getBrightness(value);
  const contrastRatio = getContrast(value, theme.colors.interface010);
  const contrastRating = getContrastRating(contrastRatio);

  return {
    name,
    value,
    group: getColorGroup(name),
    brightness,
    contrastRating,
    contrastRatio,
    parentColor:
      paletteColorMap && paletteColorMap[value]
        ? paletteColorMap[value][0]
        : undefined,
  } as ThemeColor;
};

export const groupColorObjects = (
  acc: Record<string, ThemeColor[]>,
  color: ThemeColor,
) => {
  // eslint-disable-next-line no-multi-assign
  const group = (acc[color.group] = acc[color.group] || []);
  group.push(color);
  return acc;
};

export const createPaletteColorMap = (paletteColors: ThemeColor[]) =>
  paletteColors.reduce(
    (acc, color) => {
      // eslint-disable-next-line no-multi-assign
      const accGroup = (acc[color.value] = acc[color.value] || []);
      accGroup.push(color);
      return acc;
    },
    {} as Record<string, ThemeColor[]>,
  );
