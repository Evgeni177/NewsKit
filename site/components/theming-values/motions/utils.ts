export const CANVAS_SIZE = 56;

const parseCubicBezier = (value: string) =>
  value
    .replace('cubic-bezier(', '')
    .replace(')', '')
    .split(',')
    .map(v => Number.parseFloat(v.trim()));

// NEXT 2 FUNCTIONS ARE COPIED FROM http://dbaron.org/css/timing-function-graphs
export function bezier(x1: number, y1: number, x2: number, y2: number) {
  // Cubic bezier with control points (0, 0), (x1, y1), (x2, y2), and (1, 1).
  function xForT(t: number) {
    const omt = 1 - t;
    return 3 * omt * omt * t * x1 + 3 * omt * t * t * x2 + t * t * t;
  }
  function yForT(t: number) {
    const omt = 1 - t;
    return 3 * omt * omt * t * y1 + 3 * omt * t * t * y2 + t * t * t;
  }
  function tForX(x: number) {
    // Binary subdivision.
    let mint = 0;
    let maxt = 1;
    for (let i = 0; i < 30; ++i) {
      const guesst = (mint + maxt) / 2;
      const guessx = xForT(guesst);
      if (x < guessx) maxt = guesst;
      else mint = guesst;
    }
    return (mint + maxt) / 2;
  }
  return function bezierClosure(x: number) {
    if (x === 0) return 0;
    if (x === 1) return 1;
    return yForT(tForX(x));
  };
}

export const updateCanvas = (
  ctx: CanvasRenderingContext2D,
  timingFn: string,
  color: string,
) => {
  let x1 = 0;
  let y1 = 0;
  let x2 = 0;
  let y2 = 0;
  switch (timingFn) {
    case 'ease':
      x1 = 0.25;
      y1 = 0.1;
      x2 = 0.25;
      y2 = 1.0;
      break;
    case 'linear':
      x1 = 0.0;
      y1 = 0.0;
      x2 = 1.0;
      y2 = 1.0;
      break;
    case 'ease-in':
      x1 = 0.42;
      y1 = 0.0;
      x2 = 1.0;
      y2 = 1.0;
      break;
    case 'ease-out':
      x1 = 0.0;
      y1 = 0.0;
      x2 = 0.58;
      y2 = 1.0;
      break;
    case 'ease-in-out':
      x1 = 0.42;
      y1 = 0.0;
      x2 = 0.58;
      y2 = 1.0;
      break;
    default:
      [x1, y1, x2, y2] = parseCubicBezier(timingFn);
      break;
  }

  const tf = bezier(x1, y1, x2, y2);

  const gCurrentTFPoints = [];

  for (let x = 0; x <= CANVAS_SIZE; ++x) {
    gCurrentTFPoints[x] = tf(x / CANVAS_SIZE) * CANVAS_SIZE;
  }

  ctx.clearRect(0, 0, CANVAS_SIZE, CANVAS_SIZE);

  // Draw the graph
  ctx.strokeStyle = color;
  ctx.lineWidth = 2;
  ctx.beginPath();
  ctx.moveTo(0, CANVAS_SIZE - gCurrentTFPoints[0]);
  for (let x = 1; x <= CANVAS_SIZE; ++x) {
    ctx.lineTo(x, CANVAS_SIZE - gCurrentTFPoints[x]);
  }
  ctx.stroke();
};

export const percentFrom = (total: number, value: number) =>
  Math.floor(value / (total / 100));

export const getTotalDuration = (stayDuration: number, maxDuration: number) =>
  2 * stayDuration + 2 * maxDuration;
