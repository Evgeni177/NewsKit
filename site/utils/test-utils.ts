import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
