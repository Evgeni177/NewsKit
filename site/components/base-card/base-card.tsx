import React from 'react';
import {Block, Headline} from 'newskit';
import {BaseCardProps} from './types';
import {StyledCard} from './styled';

export const BaseCard: React.FC<BaseCardProps> = ({
  title,
  children,
  layout,
  href,
  ...rest
}) => {
  const overrides = href
    ? {
        stylePreset: 'baseCardInteractive',
        teaserContainer: {
          spaceInset: 'space050',
        },
      }
    : {
        stylePreset: 'baseCardNonInteractive',
        mediaContainer: {
          spaceStack: layout === 'horizontal' ? 'space050' : 'space040',
        },
      };
  return (
    <StyledCard overrides={overrides} layout={layout} href={href} {...rest}>
      <Block spaceStack="space045">
        <Headline
          overrides={{
            typographyPreset: 'editorialHeadline030',
            heading: {
              stylePreset: 'inkContrast',
            },
          }}
        >
          {title}
        </Headline>
      </Block>
      {children}
    </StyledCard>
  );
};
