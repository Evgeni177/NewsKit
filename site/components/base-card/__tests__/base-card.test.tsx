import {renderToFragmentWithTheme} from '../../../utils/test-utils';
import {BaseCardProps} from '..';
import {BaseCard} from '../base-card';

describe('BaseCard', () => {
  test('Renders default (vertical) non-interactive card', () => {
    const props: BaseCardProps = {
      title: 'Non-interactive Vertical Card',
      description: 'Text',
    };
    const fragment = renderToFragmentWithTheme(BaseCard, props);
    expect(fragment).toMatchSnapshot();
  });

  test('Renders horizontal non-interactive card', () => {
    const props: BaseCardProps = {
      title: 'Non-interactive Horizontal Card',
      layout: 'horizontal',
      description: 'Text',
    };
    const fragment = renderToFragmentWithTheme(BaseCard, props);
    expect(fragment).toMatchSnapshot();
  });

  test('Renders vertical interactive card', () => {
    const props: BaseCardProps = {
      title: 'Interactive Horizontal Card',
      href: 'newskit.com',
      description: 'Text',
    };
    const fragment = renderToFragmentWithTheme(BaseCard, props);
    expect(fragment).toMatchSnapshot();
  });

  test('Renders horizontal interactive card', () => {
    const props: BaseCardProps = {
      title: 'Interactive Horizontal Card',
      href: 'newskit.com',
      layout: 'horizontal',
      description: 'Text',
    };
    const fragment = renderToFragmentWithTheme(BaseCard, props);
    expect(fragment).toMatchSnapshot();
  });
});
