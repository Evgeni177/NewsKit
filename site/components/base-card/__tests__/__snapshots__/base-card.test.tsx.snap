// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard Renders default (vertical) non-interactive card 1`] = `
<DocumentFragment>
  .emotion-6 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
}

.emotion-5 {
  box-sizing: border-box;
}

.emotion-4 {
  box-sizing: border-box;
  padding: 8px 4px;
}

.emotion-4 a:not(.nk-card-link) {
  z-index: 2;
  position: relative;
}

.emotion-2 {
  margin-bottom: 20px;
}

.emotion-1 {
  font-family: "Source Serif Pro",serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0px;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.2333333333333334em + 1px);
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.2958333333333334em + 1px);
}

.emotion-0 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #0a0a0a;
}

.emotion-0 svg {
  fill: #0a0a0a;
}

.emotion-3 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-3 svg {
  fill: #2e2e2e;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.4em;
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4625em;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <div
        class="emotion-4"
      >
        <div
          class="emotion-2"
        >
          <section
            class="emotion-1"
          >
            <h1
              class="nk-headline-heading emotion-0"
            >
              Non-interactive Vertical Card
            </h1>
          </section>
        </div>
        <p
          class="emotion-3"
        >
          Text
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BaseCard Renders horizontal interactive card 1`] = `
<DocumentFragment>
  .emotion-3 {
  margin-bottom: 20px;
}

.emotion-1 {
  font-family: "Source Serif Pro",serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0px;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.2333333333333334em + 1px);
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.2958333333333334em + 1px);
}

.emotion-0 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #0a0a0a;
}

.emotion-0 svg {
  fill: #0a0a0a;
}

.emotion-4 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-4 svg {
  fill: #2e2e2e;
}

.emotion-4::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.4em;
}

.emotion-4::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4625em;
}

.emotion-7 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: rowfalse;
  -ms-flex-direction: rowfalse;
  flex-direction: rowfalse;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-6 {
    min-width: 1px;
  }
}

.emotion-2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
}

.emotion-2 .nk-headline-kicker {
  color: #0a68c1;
}

.emotion-2:hover:not(:disabled) .nk-headline-kicker {
  color: #02509b;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:active:not(:disabled) .nk-headline-kicker {
  color: #01396e;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:visited:not(:disabled) .nk-headline-kicker {
  color: #7547a3;
}

.emotion-2 .nk-headline-heading {
  color: #0a0a0a;
}

.emotion-2 .nk-headline-heading svg {
  fill: #0a0a0a;
}

.emotion-2:before {
  content: '';
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  overflow: hidden;
  position: absolute;
  z-index: 1;
}

.emotion-5 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 24px;
}

.emotion-5 a:not(.nk-card-link) {
  z-index: 2;
  position: relative;
}

<div
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-3"
        >
          <a
            class="nk-card-link emotion-2"
            href="newskit.com"
          >
            <section
              class="emotion-1"
            >
              <h1
                class="nk-headline-heading emotion-0"
              >
                Interactive Horizontal Card
              </h1>
            </section>
          </a>
        </div>
        <p
          class="emotion-4"
        >
          Text
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BaseCard Renders horizontal non-interactive card 1`] = `
<DocumentFragment>
  .emotion-2 {
  margin-bottom: 20px;
}

.emotion-1 {
  font-family: "Source Serif Pro",serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0px;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.2333333333333334em + 1px);
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.2958333333333334em + 1px);
}

.emotion-0 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #0a0a0a;
}

.emotion-0 svg {
  fill: #0a0a0a;
}

.emotion-3 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-3 svg {
  fill: #2e2e2e;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.4em;
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4625em;
}

.emotion-6 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: rowfalse;
  -ms-flex-direction: rowfalse;
  flex-direction: rowfalse;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
}

.emotion-5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-5 {
    min-width: 1px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 8px 4px;
}

.emotion-4 a:not(.nk-card-link) {
  z-index: 2;
  position: relative;
}

<div
    class="emotion-6"
  >
    <div
      class="emotion-5"
    >
      <div
        class="emotion-4"
      >
        <div
          class="emotion-2"
        >
          <section
            class="emotion-1"
          >
            <h1
              class="nk-headline-heading emotion-0"
            >
              Non-interactive Horizontal Card
            </h1>
          </section>
        </div>
        <p
          class="emotion-3"
        >
          Text
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BaseCard Renders vertical interactive card 1`] = `
<DocumentFragment>
  .emotion-7 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
}

.emotion-6 {
  box-sizing: border-box;
}

.emotion-3 {
  margin-bottom: 20px;
}

.emotion-1 {
  font-family: "Source Serif Pro",serif;
  font-size: 24px;
  line-height: 1.1666666666666667;
  font-weight: 600;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0px;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.2333333333333334em + 1px);
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: calc(-0.2958333333333334em + 1px);
}

.emotion-0 {
  display: inline;
  font: inherit;
  margin: 0;
  color: #0a0a0a;
}

.emotion-0 svg {
  fill: #0a0a0a;
}

.emotion-4 {
  margin: 0;
  padding: 1px 0;
  color: #2e2e2e;
  font-family: "Noto Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-4 svg {
  fill: #2e2e2e;
}

.emotion-4::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.4em;
}

.emotion-4::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4625em;
}

.emotion-5 {
  box-sizing: border-box;
  padding: 24px;
}

.emotion-5 a:not(.nk-card-link) {
  z-index: 2;
  position: relative;
}

.emotion-2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
}

.emotion-2 .nk-headline-kicker {
  color: #0a68c1;
}

.emotion-2:hover:not(:disabled) .nk-headline-kicker {
  color: #02509b;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:active:not(:disabled) .nk-headline-kicker {
  color: #01396e;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:visited:not(:disabled) .nk-headline-kicker {
  color: #7547a3;
}

.emotion-2 .nk-headline-heading {
  color: #0a0a0a;
}

.emotion-2 .nk-headline-heading svg {
  fill: #0a0a0a;
}

.emotion-2:before {
  content: '';
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  overflow: hidden;
  position: absolute;
  z-index: 1;
}

<div
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-3"
        >
          <a
            class="nk-card-link emotion-2"
            href="newskit.com"
          >
            <section
              class="emotion-1"
            >
              <h1
                class="nk-headline-heading emotion-0"
              >
                Interactive Horizontal Card
              </h1>
            </section>
          </a>
        </div>
        <p
          class="emotion-4"
        >
          Text
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
